engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ejea/EjeaFetchingCachedOutputsFromDatabaseSpec.scala

```diff
@@ -0,0 +1,74 @@
+package cromwell.engine.workflow.lifecycle.execution.ejea
+
+import cromwell.engine.workflow.lifecycle.execution.EngineJobExecutionActor._
+import EngineJobExecutionActorSpec._
+import cromwell.backend.BackendCacheHitCopyingActor.CopyOutputsCommand
+import cromwell.core.callcaching.{CallCachingActivity, ReadAndWriteCache}
+import cromwell.engine.workflow.lifecycle.execution.callcaching.EngineJobHashingActor.{CacheHit, HashError}
+import cromwell.engine.workflow.lifecycle.execution.callcaching.FetchCachedResultsActor.{CachedOutputLookupFailed, CachedOutputLookupSucceeded}
+import cromwell.engine.workflow.lifecycle.execution.callcaching.MetaInfoId
+
+import scala.util.{Failure, Success}
+
+class EjeaFetchingCachedOutputsFromDatabaseSpec extends EngineJobExecutionActorSpec with HasJobSuccessResponse {
+
+  implicit override def stateUnderTest = FetchingCachedOutputsFromDatabase
+
+  "An EJEA in FetchingCachedOutputsFromDatabase state" should {
+
+    val possibleEjhaResponses = List(
+      ("no hashes", None),
+      ("hash results", Some(callCacheHashes)),
+      ("hash error", Some(HashError(new Exception("blah")))))
+
+    possibleEjhaResponses foreach { case (name, ejhaResponse) =>
+      s"Correctly receive $name then begin the backend-specific output copying actor when it gets a result-fetch success" in {
+        ejea = ejeaInFetchingCachedOutputsFromDatabaseState()
+        // Send the response from the EJHA (if there was one!):
+        ejhaResponse foreach { ejea ! _ }
+
+        // Send the response from the "Fetch" actor
+        val cachedResults = successResponse.jobOutputs
+        ejea ! CachedOutputLookupSucceeded(cachedResults, CacheHit(MetaInfoId(75)))
+        helper.callCacheHitCopyingProbe.expectMsg(CopyOutputsCommand(cachedResults))
+
+        // Check we end up in the right state:
+        ejea.stateName should be(BackendIsCopyingCachedOutputs)
+        // Check we end up with the right data:
+        val expectedData = initialData.copy(hashes = ejhaResponse map {
+          case `callCacheHashes` => Success(callCacheHashes)
```

Comment:
ToL: What do you think about finding an object (say `object HasJobSuccessResponse`) to house and uppercase your [constants](http://docs.scala-lang.org/style/naming-conventions.html#constants-values-variable-and-methods), and remove the backtick workaround?